### PR TITLE
Update Gambit, Gerbil and libraries

### PR DESCRIPTION
**Pull request by fare**
###### Motivation for this change

Upstream update to gambit, gerbil, and gerbil-utils, and new libraries: gerbil-crypto, gerbil-poo, gerbil-persist, gerbil-ethereum.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fare** [comment]
This is coming along very nicely @7c6f434c

File: pkgs/development/compilers/gerbil/gerbil-support.nix [hunk 1]
```diff
@@ -13,21 +17,25 @@ rec {
...
     gerbil ? pkgs.gerbil-unstable,
     gambit-params ? pkgs.gambit-support.stable-params,
     gerbilInputs ? [],
     buildInputs ? [],
-    softwareName ? "" } :
+    softwareName ? ""} :
     let buildInputs_ = buildInputs; in
     gccStdenv.mkDerivation rec {
-      inherit src meta pname version;
+      inherit src meta pname version gerbil-package version-path;
```
Comment (7c6f434c) [inline pkgs/development/compilers/gerbil/gerbil-support.nix:29]: Why change the inherit here?

**Comment by 7c6f434c** [comment]
(Sorry, a bit busy lately, so I saw the PR in the stream of notifications but forgot about it until it showed up in Notifications… which I also don't keep up with _well_, but at least better)

File: pkgs/development/compilers/gerbil/gerbil-support.nix (hunk 1 above)
Comment (fare) [reply pkgs/development/compilers/gerbil/gerbil-support.nix:29]: I'm now querying the gerbil-package and version-path attributes from the packages that depend on you, so create a version file with the correct dependencies on the other version files.

Comment (7c6f434c) [reply pkgs/development/compilers/gerbil/gerbil-support.nix:29]: Maybe it should be in `passthru` then? No difference for downstream use, but names with `-` in direct `mkDerivation` arguments (that tries to export things as environment variables during the build) look suspicious.

Comment (fare) [reply pkgs/development/compilers/gerbil/gerbil-support.nix:29]: I didn't know about `passthru`. Maybe `gerbil-package` is it, but `version-path` isn't. I'll have to evaluate each variable...

Comment (7c6f434c) [reply pkgs/development/compilers/gerbil/gerbil-support.nix:29]: Why `version-path` isn't it? You use is via Nix string variable expansion, not via shell environment anyway.

Comment (7c6f434c) [reply pkgs/development/compilers/gerbil/gerbil-support.nix:29]: (checked: it gets into `.drv` file but not into the actual environment, the most fragile combination, I guess…)

Comment (fare) [reply pkgs/development/compilers/gerbil/gerbil-support.nix:29]: Let's discuss it on IRC some time this week (or next week?)

Comment (fare) [reply pkgs/development/compilers/gerbil/gerbil-support.nix]: OK, I'm now using `passthru` for `gerbil-package` and `version-path`.

**Pull request merged**